include/mxnet/ndarray.h

```diff
@@ -554,6 +559,27 @@ struct NDArrayFunctionReg
     return *this;
   }
   /*!
+   * \brief set the function body to a unary NDArray function
+   *  this will also auto set the parameters correctly
+   * \param fgeneric function body to set
+   * \return ref to the registered entry, used to set properties
+   */
+  inline NDArrayFunctionReg &set_function(
+    void (*fgeneric)(NDArray **used_vars,
+                     real_t *s,
+                     NDArray **mutate_vars,
+                     const std::map<std::string, std::string>& param)) {
```

Comment:
try pass char** char**, to reduce the cost of constructing the map, note this call can have significant amount of overhead